packages/frameworks-fastify/src/index.ts

```diff
@@ -0,0 +1,209 @@
+ * import { getSession } from "@auth/fastify"
+ *
+ * export function authSession(req: Request, res: Response, next: NextFunction) {
+ *   res.locals.session = await getSession(req)
+ *   next()
+ * }
+ *
+ * app.use(authSession)
+ *
+ * // Now in your route
+ * app.get("/", (req, res) => {
+ *   const { session } = res.locals
+ *   res.render("index", { user: session?.user })
+ * })
+ * ```
+ *
+ * ## Authorization
+ * You can protect routes by checking for the presence of a session and then redirect to a login page if the session is not present.
+ * This can either be done per route, or for a group of routes using a middleware such as the following:
+ *
+ * ```ts
+ * export function authenticatedUser(
+ *   req: Request,
+ *   res: Response,
+ *   next: NextFunction
+ * ) {
+ *   const session = res.locals.session ?? (await getSession(req, authConfig))
+ *   if (!session?.user) {
+ *     res.redirect("/login")
+ *   } else {
+ *     next()
+ *   }
+ * }
+ * ```
+ *
+ * ### Per Route
+ * To protect a single route, simply add the middleware to the route as follows:
+ * ```ts title="app.ts"
+ * // This route is protected
+ * app.get("/profile", authenticatedUser, (req, res) => {
+ *   const { session } = res.locals
+ *   res.render("profile", { user: session?.user })
+ * })
+ *
+ * // This route is not protected
+ * app.get("/", (req, res) => {
+ *   res.render("index")
+ * })
+ *
+ * app.use("/", root)
+ * ```
+ * ### Per Group of Routes
+ * To protect a group of routes, define a router and add the middleware to the router as follows:
+ *
+ * ```ts title="routes/protected.route.ts"
+ * import { Router } from "express"
+ *
+ * const router = Router()
+ *
+ * router.use(authenticatedUser) // All routes defined after this will be protected
+ *
+ * router.get("/", (req, res) => {
+ *   res.render("protected")
+ * })
+ *
+ * export default router
+ * ```
+ *
+ * Then we mount the router as follows:
+ * ```ts title="app.ts"
+ * import protected from "./routes/protected.route"
+ *
+ * app.use("/protected", protected)
+ * ```
+ *
+ * @module @auth/fastify
+ */
+
+import { Auth } from "@auth/core"
+import type { AuthConfig, Session } from "@auth/core/types"
+import type { FastifyRequest, FastifyPluginAsync } from 'fastify'
+import { toWebRequest, toFastifyReply } from "./lib/index.js"
+
+export type {
+  Account,
+  DefaultSession,
+  Profile,
+  Session,
+  User,
+} from "@auth/core/types"
+
+function FastifyAuthPlugin(authConfig: Omit<AuthConfig, "raw">): FastifyPluginAsync {
+  return async (fastify, opts) => {
+    fastify.route({
+      method: ['GET', 'POST'],
+      url: '/*',
+      handler: async (request, reply) => {
+        const response = await Auth(toWebRequest(request), authConfig);
+        return toFastifyReply(response, reply);
+      }
+    });
+  };
+}
+
+export function FastifyAuth(
+  config: AuthConfig
+): ReturnType<typeof FastifyAuthPlugin> {
+  const { ...authOptions } = config
+  authOptions.secret ??= process.env.AUTH_SECRET
```

Comment:
Can you add support for the `AUTH_REDIRECT_PROXY_URL`?

See: https://github.com/nextauthjs/next-auth/pull/9657/files

Reply:
@hillac @ndom91 
I came to this thread after searching for "TypeError: fetch failed". I am working with SvelteKitAuth (0.14) and Auth Core (0.28) with Node version (v20.9.0). I am seeing this issue a lot! 
Any hacks to fix this?